src/main/java/com/hubspot/jinjava/lib/exptest/IsEscapedExpTest.java

```diff
@@ -0,0 +1,41 @@
+package com.hubspot.jinjava.lib.exptest;
+
+import com.hubspot.jinjava.doc.annotations.JinjavaDoc;
+import com.hubspot.jinjava.doc.annotations.JinjavaParam;
+import com.hubspot.jinjava.doc.annotations.JinjavaSnippet;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.objects.SafeString;
+
+@JinjavaDoc(
+  value = "Return true if the object is marked as escaped.",
+  input = @JinjavaParam(value = "object", type = "object", required = true),
+  snippets = {
+    @JinjavaSnippet(
+      code = "{% if 'test' is escaped %}\n" +
+      "      <!--this code will not render-->\n" +
+      "{% endif %}"
+    ),
+    @JinjavaSnippet(
+      code = "{% if ('test'|escape) is escaped %}\n" +
+      "      <!--this code will render-->\n" +
+      "{% endif %}"
+    ),
+    @JinjavaSnippet(
+      code = "{% if ('test'|safe) is escaped %}\n" +
+      "      <!--this code will render-->\n" +
+      "{% endif %}"
+    )
+  }
+)
+public class IsEscapedExpTest implements ExpTest {
+
+  @Override
+  public String getName() {
+    return "escaped";
+  }
+
+  @Override
+  public boolean evaluate(Object var, JinjavaInterpreter interpreter, Object... args) {
+    return var instanceof SafeString;
```

Comment:
SafeString is kind of confusing, a string being "safe" means that you trust the contents and that it won't be escaped. 
I'm wondering if we should also check the context for interpreter.getContext().isAutoEscape . 

Something like !(var instanceof SafeString) && interpreter.getContext().isAutoEscape()

Reply:
Hi, thanks for reviewing!

I have it this way because the Jinja implementation just checks for whether the value in the variable is marked safe rather than whether it actually is safe, so whether or not the output enters that `if` statement you linked might not actually be relevant. Also, the expression test is evaluated before that condition evaluates, so I think it makes sense that the test returns whether the value is marked safe rather than whether its value would get escaped.

Reply:
![Screen Shot 2020-05-26 at 9 52 28 AM](https://user-images.githubusercontent.com/12618147/82918421-b1ed5380-9f42-11ea-9ac9-e6ee493251df.png)
Here's the Jinja behavior that I modeled this off of.

Reply:
Ok thanks for explaining. I guess it makes sense otherwise we would end up with double escaped strings.